source/src/PascalCompiler/Grammar/PascalRules.h

```diff
@@ -8,214 +8,230 @@
-    {ConstExpr3,{CJ}, TreeConstructor},
-    {ConstExpr3,{EPSILON}, TreeConstructor},
-    {ConstExpr2,{CJ}, TreeConstructor},
-    {ConstExpr2,{EPSILON}, TreeConstructor},
-    {RatianalType1,{CI,RatianalType2}, RationalStart},
-    {RatianalType2,{SRSP,CI}, RationalEnd},
-    {RatianalType2,{EPSILON}, TreeConstructor},
-// VAR DONE
-    {VarSection,{RWV,VarSection1,VarSection2}, TreeConstructor},
-    {VarSection1,{IdList,SRCN,Type,SRSM}, VariableDeclarationAction},
-    {VarSection2,{VarSection1,VarSection2}, TreeConstructor},
-    {VarSection2,{EPSILON}, TreeConstructor},
-// ID LIST DONE
-    {IdList,{ID,IdList1}, FinalAppendList},
-    {IdList1,{SRCA,ID,IdList1}, AppendIdList},
-    {IdList1,{EPSILON}, InitIdList},
+    {Program, {RWPR, ID, SRSM, GlobalDeclaration, CompoundStatement, SRSP}, FinishMainFuntion},
+
+// GLOBAL DECLARATION
+    {GlobalDeclaration, {EPSILON}, TreeConstructor},
+    {GlobalDeclaration, {NotEmptyGlobalDeclaration, GlobalDeclaration}, TreeConstructor},
+    {NotEmptyGlobalDeclaration, {LabelSection}, TreeConstructor},
+    {NotEmptyGlobalDeclaration, {ConstSection}, TreeConstructor},
+    {NotEmptyGlobalDeclaration, {TypeSection}, TreeConstructor},
+    {NotEmptyGlobalDeclaration, {VarSection}, TreeConstructor},
+    {NotEmptyGlobalDeclaration, {OperatorOverloadSection}, TreeConstructor},
+    {NotEmptyGlobalDeclaration, {FunctionDeclaration}, TreeConstructor},
+    {NotEmptyGlobalDeclaration, {ProcedureDeclaration}, TreeConstructor},
+
+// LABEL SECTION (DONE)
+    {LabelSection, {RWLB, Label, NextLabel, SRSM}, LabelRegistration},
+    {NextLabel, {EPSILON}, TreeConstructor},
+    {NextLabel, {SRCA, Label, NextLabel}, LabelRegistration},
+    {Label, {ID}, LabelID},
+
+// CONST SECTION (NOT SUPPORTED)
+    {ConstSection, {RWCN, ConstDefinition, SRSM, NextConstDefinition}, TreeConstructor},
+    {NextConstDefinition, {EPSILON}, TreeConstructor},
+    {NextConstDefinition, {ConstDefinition, SRSM, NextConstDefinition}, TreeConstructor},
+    {ConstDefinition, {ID, OPEQ, Expression}, TreeConstructor}, // Expression => ConstExpression
+
+// TYPE SECTION (DONE)
+    {TypeSection, {RWTP, TypeDefinition, SRSM, NextTypeDefinition}, TreeConstructor},
+    {NextTypeDefinition, {EPSILON}, TreeConstructor},
+    {NextTypeDefinition, {TypeDefinition, SRSM, NextTypeDefinition}, TreeConstructor},
+    {TypeDefinition, {ID, OPEQ, Type}, TypeRegistrationAction},
+    {Type, {BuiltInType}, TypePropagation},
+    {Type, {Massive}, TypePropagation},
+    {Type, {EnumType}, TypePropagation},
+    {Type, {RecordType}, TypePropagation}, // TODO RECORD
+    {Type, {ID}, GetTypeAccordingToID},
+    {DefiniteType, {BuiltInType}, TypePropagation},
+    {DefiniteType, {ID}, GetTypeAccordingToID},
+    {BuiltInType, {RWINT}, IntegerPass},
+    {BuiltInType, {RWRAT}, RationalPass},
+    {BuiltInType, {RWCX}, ComplexPass},
+    {Massive, {RWAR, SRLB, Index, NextIndex, SRRB, RWOF, DefiniteType}, ArrayCreationAction},
```

Comment:
I think, that we should consider about using empty index list as pointer to array.